xds/internal/xdsclient/xdsresource/type_eds.go

```diff
@@ -49,9 +49,10 @@ const (
 
 // Endpoint contains information of an endpoint.
 type Endpoint struct {
-	Address      string
-	HealthStatus EndpointHealthStatus
-	Weight       uint32
+	Address           string
+	AdditionalAddress []string
```

Comment:
I wasn't concerned about the amount of changes  because the compiler would flag all the places that needed to be changed. The reasons why I added a new field were:
1. The change would be backward incompatible, probably why envoy also added a list for additional addresses.
2. For a valid Endpoint, the `Address` should always be present, the Additional addresses may be empty or not. When we change to have only one `Addresses` list, this constraint now becomes that the address list size is guaranteed to be > 1 and the first address of the list is the primary address. The special status of the first address seems like it its better represented by having it stored separately.

I'm open to changing this to a single string slice, I would like to hear your thoughts on the above.

Reply:
> The change would be backward incompatible, probably why envoy also added a list for additional addresses.

Yes, it would be a backward incompatible change if it was made in the proto. But this is an internal struct for us, and we can do whatever makes the most sense.

> The special status of the first address seems like it its better represented by having it stored separately.

In what way is the first address special?